### PR TITLE
docs: Add v7 migration guide

### PR DESCRIPTION
**Pull request by mattsjones**

### CODE REVIEW

File: docs/migration-guides/v7.0.0.md [hunk 1]
```diff
@@ -0,0 +1,289 @@
+# v7.0.0 Migration guide
+
+This release changes the way sku handles rendering html & asset generation for both server rendered and statically rendered apps.
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
This release changes the way sku handles HTML rendering and asset generation for both server rendered and statically rendered apps.
```

File: docs/migration-guides/v7.0.0.md [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+Please read through [static-rendering](../static-rendering) before attempting migration.
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
Please read through our newly added documentation on [static rendering](../static-rendering) before attempting migration.
```

File: docs/migration-guides/v7.0.0.md [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+
+## Static apps
+
+Build output folder structure has changed in some circumstances. After making the following changes, please check your `target` directory and update any testing and deployment strategies to suit. See [static-rendering](../static-rendering) for details on the new structure.
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
Build output folder structure has changed in some circumstances. After making the following changes, please check your `target` directory and update any testing and deployment strategies to suit. See our documentation on [static rendering](../static-rendering) for details on the new structure.
```

File: docs/migration-guides/v7.0.0.md [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+
+### render
+
+If your app renders multiple pages/routes then first open up your `sku.config.js`.
+Using the new `routes` option, detail all your applications routes.
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
Using the new `routes` option, detail all of your application's routes.
```

File: docs/migration-guides/v7.0.0.md [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+
+Please read through [static-rendering](../static-rendering) before attempting migration.
+
+## Static apps
+
+Build output folder structure has changed in some circumstances. After making the following changes, please check your `target` directory and update any testing and deployment strategies to suit. See [static-rendering](../static-rendering) for details on the new structure.
+
+### render
+
+If your app renders multiple pages/routes then first open up your `sku.config.js`.
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
If your app renders multiple pages/routes, first open up your `sku.config.js`.

```

File: docs/migration-guides/v7.0.0.md [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+Using the new `routes` option, detail all your applications routes.
+
+Also you now **must** set a `publicPath`.
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
Also, you now **must** set a `publicPath`.
```

File: docs/migration-guides/v7.0.0.md [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+
+  // return HTML for each route
+  return {
+    '/': renderHTML('/'),
+    '/details': renderHTML('/details')
+  };
+};
+```
+
+The render entry now requires two function. `renderApp`, which should render your React app. And `renderDocument` which should render the HTML document. They both will be called for each route you specify in `sku.config.js`.
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
The render entry now requires two functions:
- `renderApp`, which should render your React app.
- `renderDocument` which should render the HTML document.

They will both be called for each route you specify in `sku.config.js`.
```

File: docs/migration-guides/v7.0.0.md [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+
+Also `main.js` & `styles.css` are no longer created. Instead, render the provided `headTags` and `bodyTags` at the end of their respective sections.
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
Also, `main.js` & `styles.css` are no longer created. Instead, render the provided `headTags` and `bodyTags` at the end of their respective sections of the document.
```

File: docs/migration-guides/v7.0.0.md [hunk 9]
```diff
@@ -0,0 +1,291 @@
...
+        ${bodyTags}
+      </body>
+    </html>
+  `
+};
+```
+
+### locales
+
+The `locales` config has been removed. In most cases apps will be able to rename this property to `sites`.
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
The `locales` config has been removed. In most cases, apps will be able to rename this property to `sites`.
```

File: docs/migration-guides/v7.0.0.md [hunk 10]
```diff
@@ -0,0 +1,291 @@
...
+module.exports = {
+  locales: ['au', 'nz']
+};
+// becomes
+module.exports = {
+  sites: ['au', 'nz']
+};
+```
+
+Then in the render entry pass the site option through to the React app.
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
Then, in the render entry, pass the `site` option through to the React app.
```

File: docs/migration-guides/v7.0.0.md [hunk 11]
```diff
@@ -0,0 +1,291 @@
...
+import App from './App';
+
+hydrate(<App site={window.APP_CONFIG.site} />, document.getElementById('app'));
+```
+
+[React context](https://reactjs.org/docs/context.html) can ease the pain of forwarding config props through your app.
+
+### env
+
+The `env` config has been deprecated as it conflicts with the way sku builds assets. The `env` option allowed apps to embed environment specific config into the assets, sku now encourages apps to create environment agnostic assets, moving the config management responsibility to the HTML file.
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
The `env` config has been deprecated as it conflicts with the way sku builds assets. The `env` option allowed apps to embed environment specific config into the assets, whereas sku now encourages apps to create environment agnostic assets, moving the config management responsibility to the HTML file at run time.
```

File: docs/migration-guides/v7.0.0.md [hunk 12]
```diff
@@ -0,0 +1,291 @@
...
+const config = window.APP_CONFIG;
+
+hydrate(<App apiUrl={config.apiUrl} />, document.getElementById('app'));
+```
+
+[React context](https://reactjs.org/docs/context.html) can ease the pain of forwarding config props through your app.
+
+Breaking changes:
+
+- Build output structure has changed and could break current deployment strategies
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
- Build output structure has changed and could break current deployment strategies.
```

File: docs/migration-guides/v7.0.0.md [hunk 13]
```diff
@@ -0,0 +1,291 @@
...
+- Update SSR port config
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]: Is this meant to be here?

File: docs/migration-guides/v7.0.0.md [hunk 14]
```diff
@@ -0,0 +1,291 @@
...
+
+## Server rendered apps
+
+### port
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
### Port
```

File: docs/migration-guides/v7.0.0.md [hunk 15]
```diff
@@ -0,0 +1,291 @@
...
+  port: { client: 1234, backend: 5678 }
+};
+// becomes
+module.exports = {
+  port: 1234,
+  serverPort: 5678
+};
+```
+
+### server entry
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
### Server entry
```

File: docs/migration-guides/v7.0.0.md [hunk 16]
```diff
@@ -0,0 +1,291 @@
...
+
+This release changes the way sku handles rendering html & asset generation for both server rendered and statically rendered apps.
+
+Please read through [static-rendering](../static-rendering) before attempting migration.
+
+## Static apps
+
+Build output folder structure has changed in some circumstances. After making the following changes, please check your `target` directory and update any testing and deployment strategies to suit. See [static-rendering](../static-rendering) for details on the new structure.
+
+### render
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
### Rendering
```

File: docs/migration-guides/v7.0.0.md [hunk 17]
```diff
@@ -0,0 +1,291 @@
...
+      <body>
+        <div id="app">${app}</div>
+        ${bodyTags}
+      </body>
+    </html>
+  `
+};
+```
+
+### locales
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
### `locales`
```

File: docs/migration-guides/v7.0.0.md [hunk 18]
```diff
@@ -0,0 +1,291 @@
...
+import React from 'react';
+import { hydrate } from 'react-dom';
+import App from './App';
+
+hydrate(<App site={window.APP_CONFIG.site} />, document.getElementById('app'));
+```
+
+[React context](https://reactjs.org/docs/context.html) can ease the pain of forwarding config props through your app.
+
+### env
```
Comment (markdalgleish) [inline docs/migration-guides/v7.0.0.md]:
```suggestion
### `env`
```

File: docs/migration-guides/v7.0.0.md (hunk 13 above)
Comment (mattsjones) [reply docs/migration-guides/v7.0.0.md]: Woops

**Pull request merged**